### PR TITLE
Split up large plotting sections in example gallery

### PR DESCRIPTION
**Pull request by Cadair**
There are probably more examples worthy of this change, but these were the ones I found.

### CODE REVIEW

File: examples/plotting/wcsaxes_map_example.py [hunk 1]
```diff
@@ -28,19 +28,40 @@
...
         label=f'Pixel coordinate [{pixel_coord[0]}, {pixel_coord[1]}]')
-ax.coords.grid(color='yellow', linestyle='solid', alpha=0.5)
 
-map_coord = ([-300, 200] * u.arcsec)
+
+################################################################################
 # Using the transform command expects coordinates in degrees and not arcseconds
+
+# sphinx_gallery_defer_figures
+
```
Comment (ayshih) [inline examples/plotting/wcsaxes_map_example.py:60]:
```suggestion
```
Don't defer a figure in the last code block of an example.  You pushed it onto the first code block of the next example!

File: .gitignore [hunk 2]
```diff
@@ -220,6 +220,7 @@ tags
 sunpy_map.asdf
 baseline
 docs/guide/data_types/figure.png
+october_M1_flares.csv
```
Comment (nabobalis) [inline .gitignore:223]: I need to double check there arent others.

File: CHANGELOG.rst [hunk 3]
```diff
@@ -213,7 +213,7 @@ Features
   ``.wcs`` is accessed) (`#4467 <https://github.com/sunpy/sunpy/pull/4467>`__)
```
Comment (nabobalis) [inline CHANGELOG.rst:213]: I gave up on the changelog.

File: docs/conf.py [hunk 4]
```diff
@@ -1,10 +1,6 @@
 """
 Configuration file for the Sphinx documentation builder.
-
-isort:skip_file
 """
-# flake8: NOQA: E402
```
Comment (nabobalis) [inline docs/conf.py:6]: These didn't seem to matter in the file.

File: examples/plotting/simple_differential_rotation.py [hunk 5]
```diff
@@ -27,7 +27,7 @@
 
 ##############################################################################
 # Next lets explore solar differential rotation by replicating Figure 1
-# in Beck 1999
+# in Beck 1999.
```
Comment (nabobalis) [inline examples/plotting/simple_differential_rotation.py:30]: I am not sure what this reference was.

File: CHANGELOG.rst (hunk 3 above)
Comment (Cadair) [reply CHANGELOG.rst:213]: yeah let's no do this to the changelog lol

File: examples/differential_rotation/differentially_rotated_gridlines.py [hunk 6]
```diff
@@ -27,20 +27,17 @@
 # observer over 27 days.
 
 aiamap = sunpy.map.Map(AIA_171_IMAGE)
-
 fig = plt.figure()
 ax = plt.subplot(projection=aiamap)
 aiamap.plot(clip_interval=(1., 99.95)*u.percent)
 
 # Lines of constant longitude prior to differential rotation
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply examples/differential_rotation/differentially_rotated_gridlines.py:34]: ?

File: examples/map/difference_images.py [hunk 7]
```diff
@@ -8,8 +8,7 @@
 This basic method works for base difference or running difference. Just change whether
 you're subtracting the previous image or the first image in a sequence.
 """
-###########################################################################
-
+# sphinx_gallery_thumbnail_number = 2
 import matplotlib.colors as colors
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply examples/map/difference_images.py:12]: Nah.

File: examples/map_transformations/reprojection_aia_euvi_mosaic.py [hunk 8]
```diff
@@ -70,21 +66,18 @@
...
 ax.add_artist(circle)
 ax.text(earth.lon.to_value("rad")+0.05, earth.radius.to_value(r_unit), "Earth")
-
 for this_satellite, this_coord in [(m.observatory, m.observer_coordinate) for m in maps]:
     ax.plot(this_coord.lon.to('rad'), this_coord.radius.to(r_unit), 'o', label=this_satellite)
-
 ax.set_theta_zero_location("S")
 ax.set_rlim(0, 1.3)
-
 ax.legend()
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply examples/map_transformations/reprojection_aia_euvi_mosaic.py:79]: I will revert them.

**Review by nabobalis — APPROVED** [review]
Brilliant work.

**Pull request merged**